app/src/main/java/com/google/samples/apps/sunflower/compose/PlantDetailView.kt

```diff
@@ -0,0 +1,505 @@
+        )
+        PlantDescription(description)
+    }
+}
+
+@Composable
+private fun PlantDescription(description: String) {
+    // Issue: User input doesn't work properly, HTML links cannot be clicked - b/158088138
+    // HTML support coming to Compose - b/139320905
+    AndroidView(resId = R.layout.item_plant_description) {
+        (it as TextView).text = HtmlCompat.fromHtml(description, HtmlCompat.FROM_HTML_MODE_COMPACT)
+    }
+}
+
+/**
+ * Gives a parallax effect to the content at the top of a [VerticalScroller].
+ */
+@Composable
+private fun ParallaxEffect(
+    scrollerPosition: ScrollerPosition,
+    modifier: Modifier = Modifier,
+    content: @Composable (Modifier) -> Unit
+) {
+    val offset = scrollerPosition.value / ParallaxDelta
+    val offsetDp = with(DensityAmbient.current) { offset.toDp() }
+    content(Modifier.padding(top = offsetDp).plus(modifier))
+}
+
+/**
+ * Calculates offset FAB needs to keep aligned in the middle of the bottom of the picture.
+ *
+ * As the [Modifier.onPositioned] in the image is invoked after scrollPosition has changed,
+ * there's a frame delay.
+ */
+@Composable
+private fun getFabOffset(imageHeight: Int, scrollerPosition: ScrollerPosition): Dp {
+    return with(DensityAmbient.current) {
+        imageHeight.toDp() + (scrollerPosition.value / ParallaxDelta).toDp() - (56 / 2).dp
+    }
+}
+
+/**
+ * Hides an element on the screen leaving its space occupied.
+ * This should be replaced with the visible modifier in the future: b/158837937
+ *
+ * Disclaimer: This implementation assumes that the content is visible before hiding it.
+ */
+@Composable
+private fun Hide(hide: Boolean, content: @Composable (Modifier) -> Unit) {
+    var contentSize by state { IntSize.Zero }
+    if (hide) {
+        val (width, height) = remember(contentSize) {
+            with(DensityAmbient.current) {
+                contentSize.width.toDp() to contentSize.height.toDp()
+            }
+        }
+        Spacer(modifier = Modifier.size(width, height))
+    } else {
+        content(Modifier.onPositioned {
+            contentSize = it.size
+        })
+    }
+}
+
+// Toolbar state related classes and functions to achieve the CollapsingToolbarLayout animation
+private enum class ToolbarState { HIDDEN, SHOWN }
+
+private val ToolbarState.toolbarShown: Boolean
+    get() = this == ToolbarState.SHOWN
+
+private fun toolbarStateFromBoolean(show: Boolean): ToolbarState =
+    if (show) ToolbarState.SHOWN
+    else ToolbarState.HIDDEN
+
+private val toolbarAlphaKey = FloatPropKey()
+private val contentAlphaKey = FloatPropKey()
+
+private val toolbarTransitionDefinition = transitionDefinition {
+    state(ToolbarState.HIDDEN) {
+        this[toolbarAlphaKey] = 0f
+        this[contentAlphaKey] = 1f
+    }
+    state(ToolbarState.SHOWN) {
+        this[toolbarAlphaKey] = 1f
+        this[contentAlphaKey] = 0f
+    }
+    transition {
+        toolbarAlphaKey using tween<Float> {
```

Comment:
Why not use `physics` here to support interruption?

Reply:
Added! Thanks